docs/source/package/config.rst

```diff
@@ -2,7 +2,14 @@
 Configuration functionality
 ===========================
 
+The description of :mod:`.config` below is a quick guide to the basic structure
+of configuration settings in Sopel. Consult the :mod:`.config.types` module for
+more specific information about the available kinds of options, and
+:mod:`.config.core_section` for examples of how the different setting types are
+practically used.
+
 .. toctree::
+   :hidden:
```

Comment:
The main subject of each subpage is linked in the introductory paragraph; is that insufficient? It seems silly to put multiple links to (effectively) the same location clustered together like:

<img width="613" alt="image" src="https://github.com/sopel-irc/sopel/assets/164140/59e65269-f1d0-40eb-b720-6710dff23f3b">

I can make the links go to the pages instead easily enough (pending wording tweaks):

<img width="593" alt="image" src="https://github.com/sopel-irc/sopel/assets/164140/c6c0d1e5-b475-4f2b-bd34-3651f1e7ff6a">

Reply:
It is, because a paragraph is a good way to get an introduction, to have a smooth landing in unknown territory. But it is very ineffective when someone wants to navigate quickly from section to section: scanning a list is far easier and faster for the brain than it is finding a link in a paragraph.

And I'm talking from experience: I absolutely didn't realize there were link to subpages in the first paragraph. Especially with the light mode, they are far less noticeable than with the dark mode. The first screenshot you have is exactly what I'd prefer.

It is not silly: different presentations, in documentation, have different purpose. They are used by different readers, and the brain scan differently each format. My brain, for instance, will automatically skip any paragraph of text when I'm traversing the documentation for a specific module/function reference, and I'm focusing only on finding the appropriate menu or list. When I'm discovering a module (for example in the Python documentation), I read the paragraph, and more or less ignore the menu.